PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/creation/MySQLCreationAdvanced.java

```diff
@@ -0,0 +1,207 @@
+    @Getter
+    private AdminUsernameTextField adminUsernameTextField;
+    @Getter
+    private JPasswordField passwordField;
+    @Getter
+    private JPasswordField confirmPasswordField;
+
+    private AzurePasswordFieldInput passwordFieldInput;
+    private AzurePasswordFieldInput confirmPasswordFieldInput;
+
+    private AzureMySQLConfig config;
+
+    MySQLCreationAdvanced(AzureMySQLConfig config) {
+        super();
+        this.config = config;
+        $$$setupUI$$$(); // tell IntelliJ to call createUIComponents() here.
+        init();
+        initListeners();
+        setData(config);
+    }
+
+    private void init() {
+        passwordFieldInput = PasswordValidationUtils.generateAzurePasswordFieldInput(this.passwordField, this.adminUsernameTextField);
+        confirmPasswordFieldInput = PasswordValidationUtils.generateAzureConfirmPasswordFieldInput(this.confirmPasswordField, this.passwordField);
+    }
+
+    private void initListeners() {
+        this.subscriptionComboBox.addItemListener(this::onSubscriptionChanged);
+        this.resourceGroupComboBox.addItemListener(this::onResourceGroupChanged);
+        this.adminUsernameTextField.getDocument().addDocumentListener(generateAdminUsernameListener());
+        this.security.getAllowAccessFromAzureServicesCheckBox().addItemListener(this::onSecurityAllowAccessFromAzureServicesCheckBoxChanged);
+        this.security.getAllowAccessFromLocalMachineCheckBox().addItemListener(this::onSecurityAllowAccessFromLocalMachineCheckBoxChanged);
+    }
+
+    private void onSubscriptionChanged(final ItemEvent e) {
+        if (e.getStateChange() == ItemEvent.SELECTED && e.getItem() instanceof Subscription) {
+            final Subscription subscription = (Subscription) e.getItem();
+            this.resourceGroupComboBox.setSubscription(subscription);
+            this.serverNameTextField.setSubscription(subscription);
+        }
+    }
+
+    private void onResourceGroupChanged(final ItemEvent e) {
+        if (e.getStateChange() == ItemEvent.SELECTED && e.getItem() instanceof ResourceGroup) {
+            final ResourceGroup resourceGroup = (ResourceGroup) e.getItem();
+            this.serverNameTextField.setResourceGroup(resourceGroup);
+        }
+    }
+
+    private DocumentListener generateAdminUsernameListener() {
+        return new TextDocumentListenerAdapter() {
+            @Override
+            public void onDocumentChanged() {
+                if (!adminUsernameTextField.isPasswordInitialized()) {
+                    adminUsernameTextField.setPasswordInitialized(true);
+                }
+            }
+        };
+    }
+
+    private void onSecurityAllowAccessFromAzureServicesCheckBoxChanged(final ItemEvent e) {
+        if (e.getStateChange() == ItemEvent.SELECTED) {
+            config.setAllowAccessFromAzureServices(true);
+        } else if (e.getStateChange() == ItemEvent.DESELECTED) {
+            config.setAllowAccessFromAzureServices(false);
+        }
```

Comment:
as it's a 2-state/value checkbox, it can be simplified to
```
config.setAllowAccessFromAzureServices(e.getStateChange() == ItemEvent.SELECTED);
```